src/response.rs

```diff
@@ -13,18 +13,19 @@ pub struct Response<'a> {
     pub status: Status,
     /// The HTTP response content type.
     pub content_type: Option<ContentType>,
-    /// The HTTP response body.
-    pub body: Option<&'a mut [u8]>,
+    /// The content length.
+    pub content_length: Option<usize>,
+    header_buf: &'a mut [u8],
+    header_len: usize,
+    body_pos: usize,
 }
 
 impl<'a> Response<'a> {
-    pub async fn read<C>(conn: &mut C, rx_buf: &'a mut [u8]) -> Result<Response<'a>, Error>
-    where
-        C: Read,
-    {
+    pub async fn read_headers<C: Read>(conn: &mut C, header_buf: &'a mut [u8]) -> Result<Response<'a>, Error> {
```

Comment:
I think it should still be called 'read' and not 'read_headers', because it is just the entry point for potentially reading the entire response.